src/Common/src/System/Numerics/Hashing/Hash.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace System.Numerics.Hashing
+{
+    internal static class Hash
```

Comment:
Certainly my first thought at the bareword "hash" is a cryptographic hash/digest.  But maybe I'm just tainted by being a specialist.